app/gui/view/component-browser/component-list-panel/grid/src/entry/style.rs

```diff
@@ -128,42 +202,41 @@ impl Colors {
     pub fn from_main_color(
         network: &frp::Network,
         style_watch: &StyleWatchFrp,
-        color: &frp::Stream<color::Lcha>,
-        color_intensities: &frp::Stream<ColorIntensities>,
+        main_color: &frp::Stream<color::Lcha>,
+        style_colors: &frp::Stream<StyleColors>,
         is_dimmed: &frp::Stream<bool>,
     ) -> Self {
-        fn mix((c1, c2): &(color::Lcha, color::Lcha), coefficient: &f32) -> color::Lcha {
-            color::mix(*c1, *c2, *coefficient)
-        }
         let app_bg = style_watch.get_color(ensogl_hardcoded_theme::application::background);
-        let color_intensities = color_intensities.clone_ref();
+        let style_colors = style_colors.clone_ref();
+        let color_anim = color::Animation::new(network);
 
-        let intensity = Animation::new(network);
         frp::extend! { network
             init <- source_();
 
-            text_intensity <- color_intensities.map(|c| c.text);
-            bg_intensity <- color_intensities.map(|c| c.background);
-            hover_hg_intensity <- color_intensities.map(|c| c.hover_highlight);
-            dimmed_intensity <- color_intensities.map(|c| c.dimmed);
-            icon_strong_intensity <- color_intensities.map(|c| c.icon_strong);
-            icon_weak_intensity <- color_intensities.map(|c| c.icon_weak);
-
-            one <- init.constant(1.0);
-            let is_dimmed = is_dimmed.clone_ref();
-            intensity.target <+ is_dimmed.switch(&one, &dimmed_intensity);
             app_bg <- all_with(&app_bg, &init, |col, ()| color::Lcha::from(col));
-            app_bg_and_input <- all(&app_bg, color);
-            main <- app_bg_and_input.all_with(&intensity.value, mix);
-            app_bg_and_main <- all(&app_bg, &main);
-            background <- app_bg_and_main.all_with(&bg_intensity, mix).sampler();
-            hover_highlight <- app_bg_and_main.all_with(&hover_hg_intensity, mix).sampler();
-            text <- app_bg_and_main.all_with(&text_intensity, mix).sampler();
-            icon_weak <- app_bg_and_main.all_with(&icon_weak_intensity, mix).sampler();
-            icon_strong <- app_bg_and_main.all_with(&icon_strong_intensity, mix).sampler();
+            app_bg_and_main <- all(&app_bg, &color_anim.value);
+            dimmed <- style_colors.map(|s| color::Lcha::from(s.dimmed));
+            let is_dimmed = is_dimmed.clone_ref();
```

Comment:
Continuing: if the dimmed would be expressed by intensity, we could here just make a mix to compute `dimmed` value.

Reply:
Answered above